dask/dataframe/tests/test_dataframe.py

```diff
@@ -3132,6 +3132,16 @@ def test_astype():
     assert_eq(a.x.astype(float), df.x.astype(float))
 
 
+def test_astype_skips_if_type_unchanged():
+    df = pd.DataFrame(
+        {"x": [1.0, 2.0, 3.0, 4.0], "y": [10, 20, 30, 40]}, index=[10, 20, 30, 40]
+    )
+    a = dd.from_pandas(df, 2)
+    b = a.astype({"x": "float", "y": "int64"})
+
+    assert_eq(a, b)
```

Comment:
This test does not fail with the change removed (I thought `assert_eq` would run some task-graph checks also)

Is there an already-existing utility for me to verify the task graph for these two Dask Dataframes are the same?

Reply:
I think this will do fine here: `a.__dask_graph__() == b.__dask_graph__()`

Reply:
I recommend `assert a is b`

Reply:
🤦‍♂️ It is obvious once you say it. Thank you, test updated.